backend/src/org/akvo/lumen/component/transformation_engine.clj

```diff
@@ -11,25 +11,26 @@
   (start [this]
     (reset! running? true)
     (let [engine (future
-                   (loop [;; TODO naming?
-                          transformation (.take queue)]
-                     (when-not (identical? transformation sentinel)
-                       (let [{:keys [completion-promise tenant-conn job-id dataset-id command]} transformation]
-                         (condp = (:type command)
-                           :transformation
-                           (engine/execute-transformation completion-promise
-                                                          tenant-conn
-                                                          job-id
-                                                          dataset-id
-                                                          (:transformation command))
+                   (loop [transformation (.take queue)]
+                     (try
+                       (when-not (identical? transformation sentinel)
+                         (let [{:keys [completion-promise tenant-conn job-id dataset-id command]} transformation]
+                           (condp = (:type command)
+                             :transformation
+                             (engine/execute-transformation completion-promise
+                                                            tenant-conn
+                                                            job-id
+                                                            dataset-id
+                                                            (:transformation command))
 
```

Comment:
Not a mistake, but I can change it if we choose to.

I usually do either

```
(cond
  test-a expr-1
  test-b expr-2)
```

or

```
(cond 
  another-test-1-perhaps-long-and-complicated
  another-expression-also-possibly-a-big-expression

  test-2
  long-and-complicated-expression-for-test-2)
```